service/history/visibilityQueueTaskExecutor.go

```diff
@@ -407,19 +400,20 @@ func (t *visibilityQueueTaskExecutor) recordCloseExecution(
 	searchAttributes *commonpb.SearchAttributes,
 ) error {
 
-	// Record closing in visibility store
-	retentionSeconds := int64(0)
-	namespace := defaultNamespace
-	recordWorkflowClose := true
-
 	namespaceEntry, err := t.shard.GetNamespaceCache().GetNamespaceByID(namespaceID)
-	if err != nil && !isWorkflowNotExistError(err) {
-		return err
+	if err != nil {
+		if _, notFound := err.(*serviceerror.NotFound); !notFound {
+			return err
+		}
```

Comment:
the task processing logic will treat not found as nil

Reply:
This is why it is not returned.